### PR TITLE
Lambada

### PR DESCRIPTION
**Pull request by VictorSanh**
Added LAMBADA dataset.

A couple of points of attention (mostly because I am not sure)
- The training data are compressed in a .tar file inside the main tar.gz file. I had to manually un-tar the training file to access the examples.
- The dev and test splits don't have the `category` field so I put `None` by default.

Happy to make changes if it doesn't respect the guidelines!

Victor

### CODE REVIEW

File: datasets/lambada/lambada.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir, "lambada_development_plain_text.txt"),
+                    "split": "dev",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath, split):
+        """ Yields examples. """
+        if split == "train":
```
Comment (patrickvonplaten) [inline datasets/lambada/lambada.py:119]: I think we usually want to extract in the `_split_generators` function. I think it's better to just hand over the `filepath` to `_generate_examples()` and then start with 

```python
if split == "train":
            recursion_pattern = f"{os.path.dirname(filepath)}/train-novels/*/*.txt"
            for idx, novel_path in enumerate(glob.iglob(recursion_pattern, recursive=True)):
```

Comment (VictorSanh) [reply datasets/lambada/lambada.py:119]: ok, let me move the un-tar in the `_split_generators`

**Review by patrickvonplaten — APPROVED** [review]

File: datasets/lambada/README.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+- expert-generated
+source_datasets:
+- extended|bookcorpus
+size_categories:
+- 10K<n<100K
+licenses:
+- cc-by-4.0
+---
+
+# Dataset Card for [Dataset Name]
```
Comment (yjernite) [inline datasets/lambada/README.md:22]:
```suggestion
# Dataset Card for LAMBADA
```

**Review by yjernite — COMMENTED** [review]
Looking good!

**Pull request merged**